libraries/chain/worker_evaluators.cpp

```diff
@@ -361,10 +369,54 @@ namespace golos { namespace chain {
 
                 _db.adjust_balance(_db.get_account(wto.author), wto.specification_cost);
 
-//
                 _db.push_virtual_operation(techspec_reward_operation(wto.author, to_string(wto.permlink), wto.specification_cost));
             }
         }
     }
 
+    void worker_assign_evaluator::do_apply(const worker_assign_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_21__1013, "worker_assign_operation");
+
+        _db.get_account(o.worker);
+
+        const auto& wto = _db.get_worker_techspec(o.techspec_author, o.techspec_permlink);
+
+        const auto& wpo = _db.get_worker_proposal(wto.worker_proposal_author, wto.worker_proposal_permlink);
+
+        if (!o.worker.size()) { // Unassign worker
+            GOLOS_CHECK_LOGIC(wpo.state == worker_proposal_state::work,
```

Comment:
Only worker or author can reset worker.
Here should be checking that `o.assigner == wto.techspec_author` or `o.assigner == wto.worker`